lux/action/temporal.py

```diff
@@ -0,0 +1,109 @@
+    recommendations : Dict[str,obj]
+            Object with a collection of visualizations that result from the Temporal action.
+    """
+    visuals = []
+    for c in ldf.columns:
+        if ldf.data_type[c] == "temporal":
+            try:
+                generated_vis = create_vis(ldf, c)
+                visuals.extend(generated_vis)
+            except (ValueError, AttributeError):
+                pass
+    recommendation = {
+        "action": "Temporal",
+        "description": "Show trends over <p class='highlight-descriptor'>time-related</p> attributes.",
+    }
+    # Doesn't make sense to generate a line chart if there is less than 3 datapoints (pre-aggregated)
+    if len(ldf) < 3:
+        recommendation["collection"] = []
+        return recommendation
+    vlist = VisList(visuals)
+    vlist.sort()
+    recommendation["collection"] = vlist
+    return recommendation
+
+
+def create_vis(ldf, col):
+    """
+    Creates and populates Vis objects for the different time-scale visualizations of the provided temporal column.
+
+    Parameters
+    ----------
+    ldf : lux.core.frame
+            LuxDataFrame with underspecified intent.
+
+    col : str
+            Name of temporal column.
+
+    Returns
+    -------
+    visuals : [Vis]
+            Collection of Vis objects.
+    """
+    visuals = []
+    converted_string = ldf[col].astype(str)
+    if converted_string.str.contains("-").any():
+        parsed_date = converted_string.str.extract(r"([0-9]{4})?-?([0-9]{2})?-?([0-9]{1,2})?")
+        valid_year, valid_month, valid_day = parsed_date.apply(lambda x: not x.isnull().all()).values
+
+        date_vis = Vis([lux.Clause(col, data_type="temporal")], source=ldf, score=4)
+        visuals.append(date_vis)
+
+        # If date components (year, month, day) were parsed, create individuals visualizations for each
+        if valid_year:
+            year_col = col + " (year)"
+            year_df = LuxDataFrame(
+                {year_col: pd.to_datetime(parsed_date[0], format="%Y").dt.strftime("%Y")}
+            )
+            year_vis = Vis([lux.Clause(year_col, data_type="temporal")], source=year_df, score=3)
+            visuals.append(year_vis)
+        if valid_month:
+            month_col = col + " (month)"
+            month_df = LuxDataFrame(
+                {month_col: pd.to_datetime(parsed_date[1], format="%m").dt.strftime("%m")}
+            )
+            month_vis = Vis([lux.Clause(month_col, data_type="temporal")], source=month_df, score=2)
+            visuals.append(month_vis)
+        if valid_day:
+            day_col = col + " (day)"
+            day_df = LuxDataFrame(
+                {day_col: pd.to_datetime(parsed_date[2], format="%d").dt.strftime("%d")}
+            )
+            day_vis = Vis([lux.Clause(day_col, data_type="temporal")], source=day_df, score=1)
+            visuals.append(day_vis)
+    else:
+        single_vis = Vis([lux.Clause(col, data_type="temporal")], source=ldf, score=5)
```

Comment:
How does this scoring behave when there are more than two temporal attributes? Do we show the unseparated overall first, then move onto all `(year)`, `(month )`, etc? Perhaps add an example of this in the use cases.

Reply:
The answer to this can be included into the long description.

Reply:
The scoring will result in the overall visualizations first, then all the `(year)` visualizations, all the `(month)` visualizations, etc. Is there an alternate ordering that should be considered? I'll also add an example for this situation to the testing and use cases and update the long description with the scoring behavior.

Reply:
Just popping in (: In my opinion, the current ordering can probably be improved. We have a suite of functions in `interestingness.py` that give you a sense of how visualizations are ranked. Perhaps `weighted_correlation` is a good place to start? This doesn't have to be in this PR, but just something to think about.